RHDEVDOCS-3243 Update privileged security context documentation to include information about pod timeouts (#38480)
openshift/openshift-docs: modules/op-executing-pipeline-run-and-task-run-with-custom-scc-and-service-account.adoc

[id=op-executing-pipeline-run-and-task-run-with-custom-scc-and-service-account_{context}]

> = Executing pipeline run and task run with custom SCC and service account

Reviewer: ```suggestion
= Executing pipeline run and task run by using a custom SCC and a custom service account
```

> When using the `pipelines-scc` security context contraint (SCC) associated with the default `pipelines` service account (SA), the pipeline run and task run pods may face timeouts. This happens because in the default `pipelines-scc` SCC, the `fsGroup.type` parameter is set to `MustRunAs`.

Reviewer 2: We should point to the reason why it can timeout (the access.redhat.com link or some bug report somewhere)

Reviewer: Just a personal preference, but I wouldn't use `(SA)` here. I don't think service account is really a long enough term to need to be abbreviated and would be less confusing if it was just spelled out each time. Up to you though :)

To avoid pod timeouts, you can create a custom SCC with the `fsGroup.type` parameter set to `RunAsAny`, and associate it with a custom service account.

[NOTE]

> ====
> As a best practice, use custom SCC and SA for pipeline runs and task runs. This approach allows greater flexibility and does not break the runs, when the defaults are modified during an upgrade.
> ====

Reviewer: ```suggestion
====
As a best practice, use custom SCC and SA for pipeline runs and task runs. This approach allows greater flexibility and does not break the runs when the defaults are modified during an upgrade.
====

```
don't need this comma IMO

> [dicrete]

Reviewer: I don't think you need `[dicrete]` for .these types of headings. I think they don't show up in the TOC anyway?

.Procedure

> . Define a custom SCC with the `fsGroup.type` parameter set to `RunAsAny`.

Reviewer: These steps should end in `:` instead of `.`

+
.Example: Custom SCC
[source,yaml]
----
apiVersion: security.openshift.io/v1
kind: SecurityContextConstraints
metadata:
  annotations:
    kubernetes.io/description: my-scc is a close replica of anyuid scc. pipelines-scc has fsGroup - RunAsAny.
  name: my-scc
allowHostDirVolumePlugin: false
allowHostIPC: false
allowHostNetwork: false
allowHostPID: false
allowHostPorts: false
allowPrivilegeEscalation: true
allowPrivilegedContainer: false
allowedCapabilities: null
defaultAddCapabilities: null
fsGroup:
  type: RunAsAny
groups:
- system:cluster-admins
priority: 10
readOnlyRootFilesystem: false
requiredDropCapabilities:
- MKNOD
runAsUser:
  type: RunAsAny
seLinuxContext:
  type: MustRunAs
supplementalGroups:
  type: RunAsAny
volumes:
- configMap
- downwardAPI
- emptyDir
- persistentVolumeClaim
- projected
- secret
----

> . Create the custom SCC.

Reviewer: ```suggestion
. Create the custom SCC:
```

same change for all steps

+
.Example: Create the `my-scc` SCC
[source,terminal]
----
$ oc create -f my-scc.yaml
----

. Create a custom service account.
+
.Example: Create a `fsgroup-runasany` service account
[source,terminal]
----
$ oc create serviceaccount fsgroup-runasany
----

. Associate the custom SCC with the custom SA.
+
.Example: Associate the `my-scc` SCC with the `fsgroup-runasany` SA
[source,terminal]
----
$ oc adm policy add-scc-to-user my-scc -z fsgroup-runasany
----
+
If you want to use the custom SA for privileged tasks, you can associate the `privileged` SCC with the custom SA by executing the following command:
+
.Example: Associate the `privileged` SCC with the `fsgroup-runasany` SA
[source,terminal]
----
$ oc adm policy add-scc-to-user privileged -z fsgroup-runasany
----

> . Use the custom service account in the pipeline run and task run.

Reviewer 2: Should be put a yaml example (or a `tkn` command) to show how to use it ?
